api/src/search/backend/load_agencies_to_index.py

```diff
@@ -72,25 +73,39 @@ def fetch_agencies(self) -> Sequence[Agency]:
         """
         return self.db_session.execute(select(Agency).options(selectinload("*"))).scalars().all()
 
-    def load_agencies(self, agencies: Sequence[Agency]) -> None:
-        logger.info("Loading agencies...")
-        active_agency_subquery = (
-            _construct_active_inner_query(Agency.agency_id)
-            .union(_construct_active_inner_query(Agency.top_level_agency_id))
+    def _get_agencies_by_status(self, status: OpportunityStatus) -> set[Agency]:
+        """Fetch agencies based on the status."""
+        agencies_subquery = (
+            _construct_active_inner_query(Agency.agency_id, status)
+            .union(_construct_active_inner_query(Agency.top_level_agency_id, status))
             .subquery()
         )
 
-        agency_id_stmt = select(active_agency_subquery).distinct()
-        active_agencies = set(self.db_session.execute(agency_id_stmt).scalars())
+        agency_id_stmt = select(agencies_subquery).distinct()
+        return set(self.db_session.execute(agency_id_stmt).scalars())
+
+    def load_agencies(self, agencies: Sequence[Agency]) -> None:
+        logger.info("Loading agencies...")
+
+        # Get agencies by opportunity status
+        posted_agencies = self._get_agencies_by_status(OpportunityStatus.POSTED)
+        forecasted_agencies = self._get_agencies_by_status(OpportunityStatus.FORECASTED)
+        closed_agencies = self._get_agencies_by_status(OpportunityStatus.CLOSED)
+        archived_agencies = self._get_agencies_by_status(OpportunityStatus.ARCHIVED)
 
         agencies_json = []
         for agency in agencies:
             logger.info(
                 "Preparing agency for upload to search index",
                 extra={"agency_id": agency.agency_id, "agency_code": agency.agency_code},
             )
+
             agency_json = SCHEMA.dump(agency)
-            agency_json["has_active_opportunity"] = agency.agency_id in active_agencies
+
+            agency_json["has_open_opportunity"] = agency.agency_id in posted_agencies
+            agency_json["has_forecasted_opportunity"] = agency.agency_id in forecasted_agencies
+            agency_json["has_closed_opportunity"] = agency.agency_id in closed_agencies
+            agency_json["has_archived_opportunity"] = agency.agency_id in archived_agencies
```

Comment:
I know this would take a bit to change from what you've done, but what if instead of 4 separate booleans, the dataset was a list of statuses? Instead we have effectively `"opportunity_statuses": ["posted", "archived"]`? Instead of passing in 4 separate booleans when doing the search request, you pass in the statuses.

I mostly ask because based on recent-ish discussions, we might want to completely change the opportunity statuses, if we add/remove one, we'd have to make a change in all of the same places. If we just change the values, then that's a much smaller change.

This also already has support for the search queries, that's effectively the same "overlap" sort of query we do for things like applicant type.